Bump prettier from 3.0.0 to 3.4.2

ybiquitous opened the pull request:
<!-- Each pull request must be associated with an open issue unless it's a documentation fix. If a corresponding issue does not exist, please create one so we can discuss the change first. -->

<!-- Please answer the following. We close pull requests that don't. -->

> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

Here is an official blog post for the latest version 3.4 of Prettier:
https://prettier.io/blog/2024/11/26/3.4.0

In addition, this change explicitly adds `prettier` to `package.json` because `prettier` is installed as a peer dependency of `@stylelint/prettier-config` and so it is not updated unless `@stylelint/prettier-config` is updated.

```sh-session
$ NODE_OPTIONS=--no-warnings npm v @stylelint/prettier-config peerDependencies
{ prettier: '>=3.0.0' }
```

Mouvedia commented:
concerning 6878b72

https://prettier.io/docs/en/options.html#experimental-ternaries
https://prettier.io/blog/2023/11/13/3.1.0

ybiquitous commented:
Good point. If we enabled the new ternary formatting, the format output would be as follows:

```shell
npm run format -- --experimental-ternaries
```

```sh-session
$ git status
On branch bump-prettier-to-3.4
Your branch is up to date with 'origin/bump-prettier-to-3.4'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .changeset/changelog-stylelint.cjs
	modified:   lib/assignDisabledRanges.mjs
	modified:   lib/createPartialStylelintResult.mjs
	modified:   lib/formatters/githubFormatter.mjs
	modified:   lib/formatters/verboseFormatter.mjs
	modified:   lib/getConfigForFile.mjs
	modified:   lib/getPostcssResult.mjs
	modified:   lib/lintPostcssResult.mjs
	modified:   lib/postcssPlugin.mjs
	modified:   lib/reportUnknownRuleNames.mjs
	modified:   lib/rules/alpha-value-notation/index.mjs
	modified:   lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs
	modified:   lib/rules/declaration-property-value-no-unknown/index.mjs
	modified:   lib/rules/font-family-name-quotes/index.mjs
	modified:   lib/rules/font-weight-notation/index.mjs
	modified:   lib/rules/keyframe-selector-notation/index.mjs
	modified:   lib/rules/lightness-notation/index.mjs
	modified:   lib/rules/media-feature-range-notation/index.mjs
	modified:   lib/rules/no-duplicate-at-import-rules/index.mjs
	modified:   lib/rules/selector-max-specificity/index.mjs
	modified:   lib/standalone.mjs
	modified:   lib/utils/addEmptyLineBefore.mjs
	modified:   lib/utils/getFileIgnorer.mjs
	modified:   lib/utils/matchesStringOrRegExp.mjs
```

For example,

```diff
diff --git a/lib/assignDisabledRanges.mjs b/lib/assignDisabledRanges.mjs
index bd148aa25..3404bc664 100644
--- a/lib/assignDisabledRanges.mjs
+++ b/lib/assignDisabledRanges.mjs
@@ -245,9 +245,9 @@ export default function assignDisabledRanges(root, result) {
 
 			if (ruleIsDisabled(ruleToDisable)) {
 				throw node.error(
-					isAllRules
-						? 'All rules have already been disabled'
-						: `"${ruleToDisable}" has already been disabled`,
+					isAllRules ?
+						'All rules have already been disabled'
+					:	`"${ruleToDisable}" has already been disabled`,
 					{
 						plugin: 'stylelint',
 					},
```

Personally, I don't believe that it is worth introducing the new format to add extra commits. But this topic should be discussed on another issue, not here.

jeddy3 reviewed (approved):
LGTM, thank you.
